src/modules/emote_menu/hooks/AutoRepositionPopover.jsx

```diff
@@ -0,0 +1,14 @@
+import {useEffect} from 'react';
+import useResize from '../../../common/hooks/Resize.jsx';
+import repositionPopover from '../../../utils/popover.js';
+
+const TOP_PADDING = 2;
+
+export default function useAutoPositionPopover(localRef, boundingQuerySelector, dependencies) {
+  function reposition() {
+    repositionPopover(localRef, boundingQuerySelector, TOP_PADDING);
+  }
+
+  useEffect(reposition, dependencies);
```

Comment:
why do we pass dependencies instead of organizing them here? we know the dependencies already